invenio_jobs/tasks.py

```diff
@@ -6,11 +6,51 @@
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """Tasks."""
+from datetime import datetime, timezone
 
 from celery import shared_task
+from invenio_db import db
 
+from invenio_jobs.models import Run, RunStatusEnum
 
-@shared_task(ignore_result=True)
-def run_job(job_id, args=None):
-    """Run a job."""
-    pass
+
+# TODO 1. Move to service? 2. Don't use kwargs?
+def update_run(run, **kwargs):
+    """Method to update and commit run updates."""
+    if not run:
+        return
+    for kw, value in kwargs.items():
+        setattr(run, kw, value)
+    db.session.commit()
+
+
+@shared_task(bind=True, ignore_result=True)
+def execute_run(self, run_id, kwargs=None):
+    """Execute and manage a run state and task."""
+    run = Run.query.filter_by(id=run_id).one_or_none()
+    task = self.app.tasks.get(run.job.task)
+    update_run(run, status=RunStatusEnum.RUNNING, started_at=datetime.now(timezone.utc))
+
+    try:
+        result = task.apply(kwargs=kwargs)
+    except Exception as e:
+        # TODO should we log the error in message?
```

Comment:
If you need to log this on sentry for visibility :

```python
current_app.logger.exception("...")
```

Reply:
we agreed on creating a ticket and removing  the TODO

Reply:
https://github.com/inveniosoftware/invenio-jobs/issues/22